onnxruntime/core/graph/contrib_ops/quantization_defs.cc

```diff
@@ -1240,5 +1240,210 @@ ONNX_MS_OPERATOR_SET_SCHEMA(
         .TypeConstraint("F", {"tensor(float16)"}, "Be compatible with float version.")
         .TypeAndShapeInferenceFunction(ONNX_NAMESPACE::propagateShapeAndTypeFromFirstInput));
 
+void convTransposeShapeInference(InferenceContext& ctx,
+                                 int input1Idx,
```

Comment:
nit: make function 'static' so it has file scope

Do we need `ONNX_NAMESPACE::` to qualify `InferenceContext`?

Is there going to be some other usage of this function? If not, calling it qlinearConvTransposeShapeInference would be clearer, and the input args to map could be hardcoded internally.

nit: QDeConvInferContext is a little obtuse. What does 'DeConv' have to do with 'ConvTranspose'? Given the scope it could be called something that says what it does like InputMappingInferenceContext.

nit: coding conventions - input1_idx and input2_idx. 

nit: when using `override` you don't need the `virtual`